pkg/asset: kube-{flannel,proxy}, calico-node: Tolerate all taints

klausenbusk opened the pull request:
kube-{flannel,proxy} need to run on every node, no matter what
the taints is. Without kube-{flannel,proxy} the node is never
ready.
I'm not sure if the node will work without calico-node, but it should
run on every node anyway.

Fix #699

squat commented:
coreosbot test this please

klausenbusk commented:
I just pushed this to production and it is working as expected.

weikinhuang commented:
Does this need to be added to the calico daemonset too?

klausenbusk commented:
> Does this need to be added to the calico daemonset too?

Definitely..

Done and rebased.

klausenbusk commented:
Are we waiting on anything, or can we merge this?

dghubble commented:
You could also add a toleration for your custom taint to the flannel or calico DaemonSet in the usual way. It feels a tad presumptuous to tolerate all possible taints, we've been wrong about broad assertions like that in the past. On the other hand I don't have a case in mind, since a NotReady node isn't so useful (maybe in debugging an issue, but its a stretch).

weikinhuang commented:
To be fair these are system level networking nodes. Is there a reason they shouldn't be running on a node?

dghubble commented:
Like I said, I don't have a concrete one in mind for flannel and calico. Some folks do run without kube-proxy for whatever reason (I'm not one of them).

The question we need to answer is not "is there a reason not to run them", its "is it ok to always run them" (onus lies on the broader assertion) because if it turns out there is some case, a user has to go find all the taints in use and edit components to re-add the explicit tolerations.

klausenbusk commented:
> You could also add a toleration for your custom taint to the flannel or calico DaemonSet in the usual way.



> because if it turns out there is some case, a user has to go find all the taints in use and edit components to re-add the explicit tolerations.

I think that most people, still want to run `kube-{flannel,proxy}, calico-node` after adding a taint. Not running `kube-{flannel,proxy}, calico-node` seems a bit like a corner case, and if people do that they is probably smart enough to tweak the tolerations them-self.

klausenbusk commented:
Any update on this?

klausenbusk commented:
Hmm.. Could someone decide if this is go or no-go? :)

dghubble commented:
I'm not opposed, but I'd feel better if upstream Calico manifests switched or indicated this was pending. They're ultimately the main upstream for these manifests.

dghubble commented:
coreosbot run e2e

klausenbusk commented:
Rebased..

klausenbusk commented:
Upstream PR: https://github.com/projectcalico/calico/pull/1443

klausenbusk commented:
New upstream PR: https://github.com/projectcalico/calico/pull/1634

klausenbusk commented:
@dghubble The upstream PR (https://github.com/projectcalico/calico/pull/1634) was just merged. I have rebased the PR and implemented it as upstream.

Can we merge this now?

dghubble commented:
ok to test

dghubble commented:
Please update the description since the conclusion was to update the tolerations, not to allow all.
